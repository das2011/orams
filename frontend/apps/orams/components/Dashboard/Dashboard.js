import React, { Component } from 'react'
import Alert from '@gov.au/page-alerts/lib/js/react.js'

export default class Dashboard extends Component {
  renderMessageBanner() {
    const { errorMessage, successMessage } = this.props
    if (successMessage)
      return (
        <Alert as="success">
          {successMessage}
        </Alert>
      )

    if (errorMessage) {
      return (
        <Alert as="error">
          {errorMessage}
        </Alert>
      )
    }

    return null
  }

  render() {
    return (
      <main>
        {this.renderMessageBanner()}
        <div>DASHBOARD</div>
      </main>
    )
  }
}
