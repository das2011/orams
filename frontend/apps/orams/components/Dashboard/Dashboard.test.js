import React from 'react'
import snapshotRenderer from 'react-test-renderer'
import Dashboard from './Dashboard'

describe('Dashboard component', () => {
  it('renders correctly when error message is present', () => {
    const props = {
      errorMessage: 'This is an error'
    }
    const tree = snapshotRenderer.create(<Dashboard {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when success message is present', () => {
    const props = {
      successMessage: 'Successful operation'
    }
    const tree = snapshotRenderer.create(<Dashboard {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when no banner messages present', () => {
    const props = {}
    const tree = snapshotRenderer.create(<Dashboard {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
