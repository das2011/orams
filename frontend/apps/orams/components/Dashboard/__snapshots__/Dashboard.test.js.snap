// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component renders correctly when error message is present 1`] = `
<main>
  <div
    className="au-body au-page-alerts   au-page-alerts--error"
    role="alert"
  >
    This is an error
  </div>
  <div>
    DASHBOARD
  </div>
</main>
`;

exports[`Dashboard component renders correctly when no banner messages present 1`] = `
<main>
  <div>
    DASHBOARD
  </div>
</main>
`;

exports[`Dashboard component renders correctly when success message is present 1`] = `
<main>
  <div
    className="au-body au-page-alerts   au-page-alerts--success"
    role="alert"
  >
    Successful operation
  </div>
  <div>
    DASHBOARD
  </div>
</main>
`;
